sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/ModelTests.groovy

```diff
@@ -0,0 +1,162 @@
+package com.azure.storage.file.datalake
+
+import com.azure.storage.file.datalake.models.AccessControlType
+import com.azure.storage.file.datalake.models.PathAccessControlEntry
+import com.azure.storage.file.datalake.models.PathPermissions
+import com.azure.storage.file.datalake.models.RolePermissions
+import spock.lang.Unroll
+
+class ModelTests extends APISpec{
+
+    def "RolePermissions creation"() {
+        when:
+        def permissions = new RolePermissions(true, true, true)
+        def other = new RolePermissions(permissions)
+
+        then:
+        permissions.read()
+        permissions.execute()
+        permissions.write()
+
+        other.read()
+        other.execute()
+        other.write()
+
+        permissions.equals(other)
+    }
+
+    @Unroll
+    def "RolePermissions parse Octal"() {
+        expect:
+        RolePermissions.parseOctal(octal as int).toSymbolic() == permission
+
+        where:
+        octal | permission
+        1     | "--x"
+        2     | "-w-"
+        4     | "r--"
+    }
+
+    @Unroll
+    def "RolePermissions parse Symbolic"() {
+        expect:
+        RolePermissions.parseSymbolic(symbol, false).toOctal() == permission
+
+        where:
+        symbol | permission
+        "--x"  | "1"
+        "-w-"  | "2"
+        "r--"  | "4"
+    }
```

Comment:
These tests don't seem to mirror each other in types. Why does `toOctal` produce a string and not a numeric value?

Reply:
toOctal means convert to a string with octal representation, suggestions on a better name are welcome

Reply:
toOctalString could clarify